src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1307,7 +1307,9 @@ public void clear(CoverageInfo info) {
         SimpleFeatureSource fs = dataStore.getFeatureSource(nativeName);
 
         // if feature type customization is there, apply it first
-        if (info.getAttributes() != null && !info.getAttributes().isEmpty()) {
+        if (info.getAttributes() != null
+                && !info.getAttributes().isEmpty()
+                && !info.getFeatureType().equals(fs.getSchema())) {
```

Comment:
This seems a fair check if a transform is required; this code must of been added since the last time cite tests were run

Reply:
Indeed, it happened later. Here we have "portion a fix", the ability to lock customized feature types remains, it just so happens that the list of attributes in CITE did not customize anything. A proper fix would require creating a TransformFeatureLocking down in gt-transform. Do you have a ticket to track this issue?

Reply:
https://osgeo-org.atlassian.net/browse/GEOS-10786

Reply:
I think the fix as provided is good; fixing regression.

Adding a TransformFeatureLocking could be a separate GeoTools improvement.

Reply:
It's an improvement over the current situation, for sure. 

The ticket, "Cannot lock Customized FeatureTypes", should be reworded, this fix does not allow locking customized types, it just makes the specific situation of the WFS 1.1 data directory work.

I cannot stress it enough, tickets are not bureaucracy, they are our public communication to users, we derive release notes and release announcements from them, and users look them up later when trying to figure out when and how to upgrade, or when a specific "fix" caused their odd usage of GeoServer break (one of the common cases of issues when upgrading in my experience, is that we fixed some bug users used to rely onto... think for example the WCS store parameter one, which turns a lenient behavior into a strict one).

Reply:
Suggestion, split into two tickets:
- Improvement: Check customized feature type to determine if transform wrapper needed
- Improvement: Add WFS-T Locking for customized feature types

Reply:
I changed the name of the 2 tickets